### DISCUSSION TITLE
These MDX global variables do not seem to be available in scope

### DISCUSSION THREAD

**Discussion by rheath-dremio** [mdx-js/mdx#2400] [Q&A] [answered]
I help to maintain a website that uses Docusaurus and MDX to build HTML from Markdown files. In preparation for migrating the site to Docusaurus 3.0, I've run the mdx-validator tool. One of the errors is this:
```
Error while compiling file docs/reference/sql/commands/nessie/alter-branch.md
Details: These MDX global variables do not seem to be available in scope: BRANCH COMMIT ERENCE REF TAG
```
The text referred to is some syntax for part of an SQL statement:
`ASSIGN { REF[ERENCE] | BRANCH | TAG | COMMIT }`

The syntax is between single tick marks because it is being used as a heading for the accompanying description, though it isn't a real heading with one or more \# characters at the front of it.

How can I prevent MDX from parsing these keywords as MDX global variables and generating the error message?

**Comment by ChristianMurphy** [comment, marked as answer]
Welcome @rheath-dremio! :wave:
Sorry you ran into some confusion.

> How can I prevent MDX from parsing these keywords as MDX global variables and generating the error message?

There are quite a few options, the two I'd recommend:
1. add `\` escapes before the `{` to ensure they are not interpreted as a JavaScript expression.
2. or, wrap the whole statement as inline code or as a code block with `` ` `` or ` ``` `

**Reply by rheath-dremio** [reply]
Thanks, but neither of the options that you suggest seem to work. As you can see in the screenshot, one instance of those keywords is in ``` marks, while a `\` precedes the braces where the other instance appears. However, I still get the error message. BTW, I removed the single tick marks from the heading so that the \ doesn't appear in the built HTML.
<img width="614" alt="Screenshot 2023-11-13 at 1 18 03 PM" src="https://github.com/mdx-js/mdx/assets/89795821/d74c220d-b065-4d31-b690-1e754862f724">

**Reply by ChristianMurphy** [reply]
@rheath-dremio can you test in the https://mdxjs.com/playground/ first?
What you are describing I can't replicate, it sounds like you have a non-MDX tool or a plugin in the mix which may be breaking things.

**Reply by rheath-dremio** [reply]
You're right. The problem does seem to be in our environment. The output in the playground is looking as expected, though I had to remove some of our idiosyncratic tags because I couldn't define them in the playground. 
<img width="1353" alt="Screenshot 2023-11-13 at 1 47 29 PM" src="https://github.com/mdx-js/mdx/assets/89795821/f6b8c56e-3412-4f37-85b1-5647d9df7f14">
Thanks for your help! I guess I'll have to take it from here.
